File: frontend/src/types/shipment.ts
export interface Shipment {
  id: string
  order_id: string
  platform_account_id?: string | null
  platform?: string | null
  platform_account_name?: string | null
  order_number?: string | null
  order_status?: string | null
  buyer_name?: string | null
  fulfillment_deadline_at?: string | null
  fulfillment_exception?: {
    status?: string
    severity?: string
    reasons?: string[]
    deadline_at?: string | null
    data_gaps?: string[]
  } | null
  tracking_number?: string | null
  carrier?: string | null
  shipping_method?: string | null
  status: string
  actual_weight_g?: number | null
  volumetric_weight_g?: number | null
  shipping_cost?: number | null
  origin_address?: ShipmentAddress | null
  destination_address?: ShipmentAddress | null
  estimated_delivery_date?: string | null
  actual_delivery_date?: string | null
  tracking_events?: TrackingEvent[] | null
  created_at?: string | null
}

export interface ShipmentAddress {
  market?: string
  country?: string
  city?: string
  address?: string
}

export interface TrackingEvent {
  timestamp: string
  location: string
  status: string
  description: string
}

File: backend/app/schemas/shipment.py
from datetime import datetime, date
from typing import Optional
from pydantic import BaseModel, ConfigDict


class ShipmentCreate(BaseModel):
    order_id: str
    carrier: str
    shipping_method: Optional[str] = None
    tracking_number: Optional[str] = None
    actual_weight_g: Optional[float] = None
    volumetric_weight_g: Optional[float] = None
    shipping_cost: Optional[float] = None
    origin_address: Optional[dict] = None
    destination_address: Optional[dict] = None
    estimated_delivery_date: Optional[date] = None
    status: str = "draft"


class ShipmentUpdate(BaseModel):
    carrier: Optional[str] = None
    shipping_method: Optional[str] = None
    tracking_number: Optional[str] = None
    status: Optional[str] = None
    actual_weight_g: Optional[float] = None
    volumetric_weight_g: Optional[float] = None
    shipping_cost: Optional[float] = None
    origin_address: Optional[dict] = None
    destination_address: Optional[dict] = None
    estimated_delivery_date: Optional[date] = None
    actual_delivery_date: Optional[date] = None


class ShipmentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    order_id: str
    tracking_number: Optional[str] = None
    carrier: Optional[str] = None
    shipping_method: Optional[str] = None
    status: str
    actual_weight_g: Optional[float] = None
    volumetric_weight_g: Optional[float] = None
    shipping_cost: Optional[float] = None
    origin_address: Optional[dict] = None
    destination_address: Optional[dict] = None
    estimated_delivery_date: Optional[date] = None
    actual_delivery_date: Optional[date] = None
    tracking_events: Optional[list] = None
    created_at: Optional[datetime] = None

class BatchShipmentCreate(BaseModel):
    order_ids: list[str]
    carrier: str
    shipping_method: Optional[str] = None
